---
title: "Unveiling the impact of bacteriophages targeting Shiga toxin-encoding Escherichia coli on the colon microbiome in vitro fermentation"
subtitle: "Microbial composition and diversity"
author: "Sudarshan and Gra√ßa Pinto"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document: 
    toc: yes
    toc_depth: 2
    toc_float: true
    code_folding: hide  
editor_options: 
  chunk_output_type: console
---


## A brief overview of the experiment  
Fecal samples were collected from three individuals. *In-vitro* fermentation experiment was carried out in duplicates for each condition:  
*  Control (No treatment)                           
*  Perturbation with E.coli CECT 5947      
*  Perturbation with Phage Ace           
*  Perturbation with Phage Ace and E.coli CECT 5947

We investigate effect of these on microbiota composition and function using wgs metagenomics.  
Compositional investigation was done using two tools:  
*  IGCsearch  https://github.com/snayfach/IGGsearch and https://github.com/snayfach/IGGdb  
*  MetaPhlan (v2) http://huttenhower.sph.harvard.edu/metaphlan2   

Downstream analysis is primarily done using `phyloseq`, `microbiome` and `microbiomeutilities`.    

# Pre-process  
## Load libraries  
```{r load-libs, message=FALSE, warning=FALSE}
library(data.table)
library(ggplot2)
library(purrr)
library(dplyr)
library(readr)
library(tidyr)
library(reshape2)
library(phyloseq)
library(microbiome)
library(microbiomeutilities)
library(patchwork)
library(tibble)
library(gghalves)
library(ggplot2)
library(ggpubr)
```

```{r}
#dir.create("fig_composition")

source("codes/compare_distances.R")
```

# Microbial community composition   

## IGC  

### Input IGC profiles    

Create a file that contains the OTU abundance distributed by sample
```{r read-species-profiles} 

control_files <- list.files(path="data/raw/igc/",
  pattern = "*_profile.tsv", full.names = T, recursive = FALSE)

#map(control_files, fread, select = c("species_id","species_abund")) 
#out <- map(control_files, fread, select = c("species_id","species_abund")) 
file_nam = gsub("data/raw/igc/","", control_files)

ans <- map2(control_files, 
            file_nam, 
            ~fread(.x,select = c("species_id","species_abund")) %>% 
              mutate(id = .y))
#ans

df <- bind_rows(ans)
#head(df) # check file
df_wide <- df %>% 
  reshape2::dcast(species_id  ~ id, value.var = "species_abund")
   #dcast(species_id ~ id, "species_abund")

#head(df_wide)

colnames(df_wide) = gsub("_species_profile.tsv","", colnames(df_wide))
rownames(df_wide) <- df_wide$species_id
colnames(df_wide) <- gsub("t24", "", colnames(df_wide))

df_wide[is.na(df_wide)] = 0
df_wide <- df_wide %>% 
  select(-species_id)

# make otu table  
otu_tb <- otu_table(df_wide, taxa_are_rows = T)

# These are % values. We divide by 100 to get proportions 
otu_tb <- otu_tb/100
#head(otu_tb)
```


### Input a taxonomy table 

Read the taxonomy of GTDBK for these OTU
```{r taxonomy-table}
tax_tabl <- read.csv(file = "data/raw/gtdbk_taxonomy_all_igcdb.csv", 
                     header = T, 
                     quote = "", 
                     sep = ",", 
                     stringsAsFactors = F)

rownames(tax_tabl) <- tax_tabl$species_id
tax_tabl <- tax_tabl %>% 
  select(-species_id)
tax_tab <- tax_tabl[c("Domain","Phylum","Class","Order","Family", "Genus","Species")]
#rownames(tax_tab) <- tax_tab$species_id
tax_mt <-as.matrix(tax_tab)
```

### Input metadata

```{r metadata}
meta_tabl <- read.table(file = "data/metadata_v1.txt", 
                        header = T, 
                        quote = "", 
                        sep = "\t", stringsAsFactors = F)
rownames(meta_tabl) <- meta_tabl$AssayIdentifier

STUDY <- c("D1", "D1", "D1", "D1", "D1", "D1", "D1", "D1",
           "D2", "D2", "D2", "D2", "D2", "D2", "D2", "D2",
           "D3", "D3", "D3", "D3", "D3", "D3", "D3", "D3")
meta_tabl$STUDY <- STUDY

meta_tabl$Perturbation <-  gsub( "Perturbation ", "",meta_tabl$Condition)


```

### Build phyloseq   
```{r pseq-1}
ps.igc <- merge_phyloseq(otu_tb,
                     tax_table(tax_mt),
                     sample_data(meta_tabl))
ps_met <- meta(ps.igc) %>%
  separate(SampleIdentifier, c("Sample", "Time"), "t") %>% 
  as.data.frame()
rownames(ps_met) <- rownames(meta(ps.igc))

head(ps_met)
sample_data(ps.igc) <- sample_data(ps_met) 

table(ps_met$Condition,ps_met$STUDY)


saveRDS(ps.igc, "data/rds/ps.igc.rds")
```


## Metaphlan  
### Input data  
```{r message=FALSE}

met_in <- as.data.frame(read_tsv("data/raw/metaphlan/humann2_bugslist.tsv"))

colnames(met_in) <- gsub("_prefix.assembled_metaphlan_bugs_list", "",colnames(met_in))
colnames(met_in) <- gsub("New.assembled_metaphlan_bugs_list", "",colnames(met_in))
colnames(met_in)[1] <- "ID"
rownames(met_in) <- met_in$ID
head(met_in)
colnames(met_in)
met_in <- met_in[,-1]

xnames = rownames(met_in)
shortnames = gsub(paste0(".+\\", "|"), "", xnames)
rownames(met_in) = shortnames

x2 = strsplit(xnames, split="|", fixed=TRUE)
taxmat = matrix(NA, ncol=max(sapply(x2, length)), nrow=length(x2))
colnames(taxmat) = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Strain")[1:ncol(taxmat)]
rownames(taxmat) = rownames(met_in)
for (i in 1:nrow(taxmat)){
  taxmat[i, 1:length(x2[[i]])] <- x2[[i]]
}
taxmat <- gsub("[a-z]__", "", taxmat)
taxmat <- phyloseq::tax_table(taxmat)
otutab <- phyloseq::otu_table(met_in, taxa_are_rows=TRUE)
otutab <- otutab/100


res <- phyloseq::phyloseq(taxmat, otutab)

```

### Create phyloseq  

```{r}

ps.mpn <- phyloseq::phyloseq(taxmat, otutab, phyloseq::sample_data(meta_tabl))

mpn_meta <- meta(ps.mpn) %>%
  separate(SampleIdentifier, c("Sample", "Time"), "t") %>% 
  as.data.frame()
rownames(mpn_meta) <- rownames(meta(ps.mpn))

sample_data(ps.mpn) <- sample_data(ps_met) 
table(mpn_meta$Condition,ps_met$STUDY)

ps.mpn <- subset_taxa(ps.mpn, !is.na(Species))
unique(tax_table(ps.mpn)[,"Kingdom"])
saveRDS(ps.mpn, "data/rds/ps.mpn.rds")
```


We have created two phyloseq object one with IGC profiles and other with metaphlan profiles.  


## Heatmap  
Investigate the relative abundances of top taxa at different taxonomic levels.  
For IGC no virus are annotated.  
For metaphlan there is viral annotation.  
### Domain  
```{r fig.height=4, fig.width=8}
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)

simple_heatmap(ps.mpn,
               group.facet = "Perturbation",
               group.order = NULL,
               abund.thres = 0.001,
               prev.thres = 0.01,
               level = "Kingdom",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange=TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("02_composition/figs/heatmap_metaphln_kingdom.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_kingdom.png", height = 6, width = 9, dpi=300)
```


### Phylum        
#### IGC  
```{r heatmap-phylum, message=FALSE}

heatmap.igc.phylum <- simple_heatmap(ps.igc,
                                     group.facet = "Perturbation",
                                     group.order = NULL,
                                     abund.thres = 0.001,
                                     prev.thres = 0.01,
                                     level = "Phylum",
                                     scale.color = "log10",
                                     na.fill = "white",
                                     color.fill = heat.cols,
                                     taxa.arrange=TRUE,
                                     remove.other=TRUE,
                                     panel.arrange="grid",
                                     ncol=NULL,
                                     nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))
heatmap.igc.phylum

ggsave("02_composition/figs/heatmap_igc_phylum.pdf", height = 4, width = 9)
ggsave("02_composition/figs/heatmap_igc_phylum.png", height = 4, width = 9, dpi=300)

```


#### Metaphlan    
```{r}
heatmap.mpn.phylum <- simple_heatmap(ps.mpn,
                                     group.facet = "Perturbation",
                                     group.order = NULL,
                                     abund.thres = 0.001,
                                     prev.thres = 0.01,
                                     level = "Phylum",
                                     scale.color = "log10",
                                     na.fill = "white",
                                     color.fill = heat.cols,
                                     taxa.arrange=TRUE,
                                     remove.other=TRUE,
                                     panel.arrange="grid",
                                     ncol=NULL,
                                     nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))
ggsave("02_composition/figs/heatmap_metaphln_phylum.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_phylum.png", height = 6, width = 9, dpi=300)
```

```{r fig.height=6, fig.width=10}
(heatmap.igc.phylum + labs(subtitle = "IGC"))/ (heatmap.mpn.phylum + labs(subtitle = "Metaphlan"))
ggsave("02_composition/figs/heatmap_igc_metaphln_phylum.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_metaphln_phylum.png", height = 6, width = 9, dpi=300)
```


### Class   
#### IGC  
```{r heatmap-class, message=FALSE}

heatmap.igc.class <- simple_heatmap(ps.igc,
                                    group.facet = "Perturbation",
                                    group.order = NULL,
                                    abund.thres = 0.001,
                                    prev.thres = 0.01,
                                    level = "Class",
                                    scale.color = "log10",
                                    na.fill = "white",
                                    color.fill = heat.cols,
                                    taxa.arrange=TRUE,
                                    remove.other=TRUE,
                                    panel.arrange="grid",
                                    ncol=NULL,
                                    nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))
heatmap.igc.class 

ggsave("02_composition/figs/heatmap_igc_class.pdf", height = 4, width = 9)
ggsave("02_composition/figs/heatmap_igc_class.png", height = 4, width = 9, dpi=300)

```


#### Metaphlan   

### Class    
```{r}
heatmap.mpn.class <- simple_heatmap(ps.mpn,
                             group.facet = "Perturbation",
                             group.order = NULL,
                             abund.thres = 0.001,
                             prev.thres = 0.01,
                             level = "Class",
                             scale.color = "log10",
                             na.fill = "white",
                             color.fill = heat.cols,
                             taxa.arrange=TRUE,
                             remove.other=TRUE,
                             panel.arrange="grid",
                             ncol=NULL,
                             nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))
heatmap.mpn.class
ggsave("02_composition/figs/heatmap_metaphln_class.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_class.png", height = 6, width = 9, dpi=300)
```


```{r  fig.height=6, fig.width=10}
(heatmap.igc.class + labs(subtitle = "IGC"))/(heatmap.mpn.class+ labs(subtitle = "Metaphlan"))
ggsave("02_composition/figs/heatmap_igc_metaphln_class.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_metaphln_class.png", height = 6, width = 9, dpi=300)
```

### Family      

#### IGC  
```{r heatmap-family, message=FALSE}


heatmap.igc.family <- simple_heatmap(ps.igc,
                                     group.facet = "Perturbation",
                                     group.order = NULL,
                                     abund.thres = 0.001,
                                     prev.thres = 0.01,
                                     level = "Family",
                                     scale.color = "log10",
                                     na.fill = "white",
                                     color.fill = heat.cols,
                                     taxa.arrange=TRUE,
                                     remove.other=TRUE,
                                     panel.arrange="grid",
                                     ncol=NULL,
                                     nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))


ggsave("02_composition/figs/heatmap_igc_family.pdf", height = 4, width = 9)
ggsave("02_composition/figs/heatmap_igc_family.png", height = 4, width = 9, dpi=300)

```

#### Metaphlan      
```{r}
heatmap.mpn.family <- simple_heatmap(ps.mpn,
                                     group.facet = "Perturbation",
                                     group.order = NULL,
                                     abund.thres = 0.001,
                                     prev.thres = 0.01,
                                     level = "Family",
                                     scale.color = "log10",
                                     na.fill = "white",
                                     color.fill = heat.cols,
                                     taxa.arrange=TRUE,
                                     remove.other=TRUE,
                                     panel.arrange="grid",
                                     ncol=NULL,
                                     nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("02_composition/figs/heatmap_metaphln_family.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_family.png", height = 6, width = 9, dpi=300)
```

```{r  fig.height=8, fig.width=10}
(heatmap.igc.family+ labs(subtitle = "IGC"))/(heatmap.mpn.family+ labs(subtitle = "Metaphlan"))
ggsave("02_composition/figs/heatmap_igc_metaphln_family.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_metaphln_family.png", height = 6, width = 9, dpi=300)
```

### Genus  
#### IGC  
```{r heatmap-1, message=FALSE}


heatmap.igc.genus <- simple_heatmap(ps.igc,
                                    group.facet = "Perturbation",
                                    group.order = NULL,
                                    abund.thres = 0.001,
                                    prev.thres = 0.01,
                                    level = "Genus",
                                    scale.color = "log10",
                                    na.fill = "white",
                                    color.fill = heat.cols,
                                    taxa.arrange=TRUE,
                                    remove.other=TRUE,
                                    panel.arrange="grid",
                                    ncol=NULL,
                                    nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))


ggsave("02_composition/figs/heatmap_igc_genus.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_genus.png", height = 6, width = 9, dpi=300)

```

#### Metaphlan      
```{r}
heatmap.mpn.genus <- simple_heatmap(ps.mpn,
                                    group.facet = "Perturbation",
                                    group.order = NULL,
                                    abund.thres = 0.001,
                                    prev.thres = 0.01,
                                    level = "Genus",
                                    scale.color = "log10",
                                    na.fill = "white",
                                    color.fill = heat.cols,
                                    taxa.arrange=TRUE,
                                    remove.other=TRUE,
                                    panel.arrange="grid",
                                    ncol=NULL,
                                    nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("02_composition/figs/heatmap_metaphln_genus.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_genus.png", height = 6, width = 9, dpi=300)
```

```{r  fig.height=12, fig.width=14}
(heatmap.igc.genus+ labs(subtitle = "IGC"))/(heatmap.mpn.genus+ labs(subtitle = "Metaphlan"))
ggsave("02_composition/figs/heatmap_igc_metaphln_genus.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_metaphln_genus.png", height = 6, width = 9, dpi=300)
```

*Bifidobacterium* and *Lactobacillus* are dominant taxa in each of the treatments.  

### Species  
#### IGC  

```{r}
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
heatmap.igc.species <- simple_heatmap(ps.igc,
                                      group.facet = "Perturbation",
                                      group.order = NULL,
                                      abund.thres = 0.001,
                                      prev.thres = 0.01,
                                      level = "Species",
                                      scale.color = "log10",
                                      na.fill = "white",
                                      color.fill = heat.cols,
                                      taxa.arrange=TRUE,
                                      remove.other=TRUE,
                                      panel.arrange="grid",
                                      ncol=NULL,
                                      nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))
ggsave("02_composition/figs/heatmap_igc_species.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_species.png", height = 6, width = 9, dpi=300)
```

#### Metaphlan    
```{r}
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
heatmap.mpn.species <- simple_heatmap(ps.mpn,
                            group.facet = "Perturbation",
                            group.order = NULL,
                            abund.thres = 0.001,
                            prev.thres = 0.01,
                            level = "Species",
                            scale.color = "log10",
                            na.fill = "white",
                            color.fill = heat.cols,
                            taxa.arrange=TRUE,
                            remove.other=TRUE,
                            panel.arrange="grid",
                            ncol=NULL,
                            nrow=NULL) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("02_composition/figs/heatmap_metaphln_species.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_metaphln_species.png", height = 6, width = 9, dpi=300)

#unique(spec_heat$data$Species)

```

```{r  fig.height=16, fig.width=16}
(heatmap.igc.species + labs(subtitle = "IGC"))/(heatmap.mpn.species + labs(subtitle = "Metaphlan"))
ggsave("02_composition/figs/heatmap_igc_metaphln_species.pdf", height =6, width = 9)
ggsave("02_composition/figs/heatmap_igc_metaphln_species.png", height = 6, width = 9, dpi=300)
```

## Check abundance E. coli  

### IGC  
```{r fig.height=4, fig.width=6}
mycols <- c(Control= "brown3", 
            E.coli= "steelblue",
            Phage = "grey50", 
            Phage_E.coli = "#7c65a9")

sample_data(ps.igc)$Perturbation <-  gsub( "Perturbation ", "",sample_data(ps.igc)$Condition)

grp_order <-c("Control", "E.coli", "Phage", "Phage_E.coli")

ps2 <- tax_glom(ps.igc, "Species")
taxa_names(ps2) <- make.unique(tax_table(ps2)[,"Species"])
select.taxa <- c("s__Escherichia coli")

taxa_names(ps2)[1:4]

p <- plot_listed_taxa(ps2, select.taxa, 
                      group= "Perturbation",
                      group.order = grp_order,
                      group.colors = mycols,
                      add.violin = TRUE,
                      violin.opacity = 0.3,
                      dot.opacity = 0.25,
                      box.opacity = 0.25,
                      panel.arrange= "grid")
p
```


### Metaphlan   
```{r}

mycols <- c(Control= "brown3", 
            E.coli= "steelblue",
            Phage = "grey50", 
            Phage_E.coli = "#7c65a9")

sample_data(ps.mpn)$Perturbation <-  gsub( "Perturbation ", "",sample_data(ps.mpn)$Condition)

grp_order <-c("Control", "E.coli", "Phage", "Phage_E.coli")

ps2 <- tax_glom(ps.mpn, "Species")
taxa_names(ps2) <- tax_table(ps2)[,"Species"]
select.taxa <- c("Escherichia_coli", "Escherichia_unclassified")

taxa_names(ps2)[1:4]

p <- plot_listed_taxa(ps2, select.taxa, 
                      group= "Perturbation",
                      group.order = grp_order,
                      group.colors = mycols,
                      add.violin = TRUE,
                      violin.opacity = 0.3,
                      dot.opacity = 0.25,
                      box.opacity = 0.25,
                      panel.arrange= "grid")
p
ggsave("02_composition/figs/ecoli_metaphln_species.pdf", height =6, width = 9)
ggsave("02_composition/figs/ecoli_metaphln_species.png", height = 6, width = 9, dpi=300)

```


## Effect on alpha diversity  

### Richness  
This analysis was performed as in the tutorial from https://microbiome.github.io/tutorials/
```{r alpha-div, message=FALSE}

library(gghalves)

grp_order <-c("Control", "E.coli", "Phage", "Phage_E.coli")

p.obs <- plot_diversity_stats(ps.igc, 
                              group = "Perturbation", 
                              index = "observed", 
                              group.order = grp_order, 
                              group.colors = mycols,
                              label.format="p.format",
                              stats = TRUE) + 
  ggtitle("IGC") +
  ylab("Observed Taxa") + xlab("") #+ coord_flip()

p.obs

```

```{r}
p.obs.mpn <- plot_diversity_stats(ps.mpn, 
                                  group = "Perturbation", 
                                  index = "observed", 
                                  group.order = grp_order, 
                                  group.colors = mycols,
                                  label.format="p.format",
                                  stats = TRUE) + 
  ggtitle("Metaphlan") +
  ylab("Observed Taxa") + xlab("") #+ coord_flip()

p.obs.mpn

p.obs.mpn.new <- plot_diversity_stats(ps.mpn, 
                                  group = "Perturbation", 
                                  index = "observed", 
                                  group.order = grp_order, 
                                  group.colors = mycols,
                                  label.format="p.format",
                                  stats = TRUE)+
  ylab("Observed Taxa") + xlab("") #+ coord_flip()

p.obs.mpn.new

```


```{r fig.width=9, fig.height=4}
p.obs + p.obs.mpn
```


### Evenness  
```{r}
p.inq <- plot_diversity_stats(ps.igc, 
                              group = "Perturbation", 
                              index = "dominance_gini", 
                              group.order = grp_order, 
                              group.colors = mycols,
                              label.format="p.format",
                              stats = TRUE) + 
  ggtitle("IGC") +
  ylab("Dominance Gini") + xlab("") #+ coord_flip()

```


```{r}
p.inq.mpn <- plot_diversity_stats(ps.mpn, 
                              group = "Perturbation", 
                              index = "dominance_gini", 
                              group.order = grp_order, 
                              group.colors = mycols,
                              label.format="p.format",
                              stats = TRUE) + 
  ggtitle("Metaphlan") +
  ylab("Dominance Gini") + xlab("") #+ coord_flip()

p.inq.mpn.new <- plot_diversity_stats(ps.mpn, 
                              group = "Perturbation", 
                              index = "dominance_gini", 
                              group.order = grp_order, 
                              group.colors = mycols,
                              label.format="p.format",
                              stats = TRUE) +
  ylab("Dominance Gini") + xlab("") #+ coord_flip()

p.inq.mpn.new

```

```{r fig.width=9, fig.height=4}
p.inq + p.inq.mpn

Combinealphaplot <- ggarrange(p.inq.mpn.new, p.obs.mpn.new,
                         labels = c("A", "B"),
                         ncol = 2, nrow = 1,
                         common.legend = TRUE, legend = "right")
Combinealphaplot
ggsave("02_composition/figs/alpha_metaphln.pdf", height =6, width = 9)
ggsave("02_composition/figs/alpha_metaphln.png", height = 6, width = 9, dpi=300)

```


We observe higher inequality when samples are treated with E. coli. Which taxa are dominating each of the groups?  

```{r}

dom_taxa.igc <- dominant_taxa(ps.igc,
                          level = "Species", 
                          group="Perturbation")
dom_taxa.igc$dominant_overview

```


```{r}

dom_taxa.mpn <- dominant_taxa(ps.mpn,
                          level = "Species", 
                          group="Perturbation")
dom_taxa.mpn$dominant_overview

```
As observed with heatmaps, Bifidobacterium dominate. Let's us check how many taxa contribute to 50% of the abundance.  

```{r}

sample_data(ps.igc)$Coverage <- coverage(ps.igc,0.95)

meta(ps.igc) %>% 
  group_by(Perturbation) %>%
  summarize(mean_coverage = mean(Coverage, na.rm = TRUE),
            stdv_coverage = sd(Coverage, na.rm = TRUE))

```

On average, 6 taxa contribute to 95% of the abundance in control and Phage group compared to 4.5 taxa in E.coli and Phage+E.coli treatment.  

```{r}

sample_data(ps.mpn)$Coverage <- coverage(ps.mpn,0.95)

meta(ps.mpn) %>% 
  group_by(Perturbation) %>%
  summarize(mean_coverage = mean(Coverage, na.rm = TRUE),
            stdv_coverage = sd(Coverage, na.rm = TRUE))

```


## Compare beta-diversity  

### Within  

#### IGC  

```{r}

ps.dist.igc <- distance(ps.igc, "bray")
beta_df_within.igc <- compare_distances(ps.igc, 
                                    dist.matrix=ps.dist.igc,
                                    method="median",
                                    group="Perturbation",
                                    na.rm=TRUE,
                                    type="within")
head(beta_df_within.igc)


```

#### Metaphlan  

```{r}

ps.dist.mpn <- distance(ps.mpn, "bray")
beta_df_within.mpn <- compare_distances(ps.mpn, 
                                    dist.matrix=ps.dist.mpn,
                                    method="median",
                                    group="Perturbation",
                                    na.rm=TRUE,
                                    type="within")
head(beta_df_within.mpn)


```


Plot
```{r}

#plot
p.igc <- ggplot(beta_df_within.igc, aes(comparison, within.median.dist)) + 
  geom_boxplot(width = 0.2, outlier.shape = NA)+ 
  geom_half_violin(
      position = position_nudge(x = 0.15, y = 0),
      alpha = 0.25, side = "r"
    ) +
  geom_jitter(aes(comparison, 
                  within.median.dist), 
              alpha=0.25, width = 0.1) + 
  theme_biome_utils() + 
 # ylim(0.6,0.90) +
  theme(axis.text.x = element_text(angle=90, hjust = 0.9)) + 
  labs(title="Median Bray-Curtis Dissimilarity (Between)") + 
  xlab("") + coord_flip() + ggtitle("IGC")
p.igc
```


```{r}

#plot
p.mpn <- ggplot(beta_df_within.mpn, aes(comparison, within.median.dist)) + 
  geom_boxplot(width = 0.2, outlier.shape = NA)+ 
  geom_half_violin(
      position = position_nudge(x = 0.15, y = 0),
      alpha = 0.25, side = "r"
    ) +
  geom_jitter(aes(comparison, 
                  within.median.dist), 
              alpha=0.25, width = 0.1) + 
  theme_biome_utils() + 
 # ylim(0.6,0.90) +
  theme(axis.text.x = element_text(angle=90, hjust = 0.9)) + 
  labs(title="Median Bray-Curtis Dissimilarity (Between)") + 
  xlab("") + coord_flip() + ggtitle("Metaphlan")

p.mpn

```

```{r fig.width=9, fig.height=4}
p.igc + p.mpn
```


### Between  
#### IGC  
```{r}
beta_df_between.igc <- compare_distances(ps.igc, 
                                     dist.matrix=ps.dist.igc,
                                     method="median",
                                     group="Perturbation",
                                     na.rm=TRUE,
                                     type="between")
head(beta_df_between.igc)


```


#### Metaphlan
```{r}
beta_df_between.mpn <- compare_distances(ps.mpn, 
                                     dist.matrix=ps.dist.mpn,
                                     method="median",
                                     group="Perturbation",
                                     na.rm=TRUE,
                                     type="between")
head(beta_df_between.mpn)


```

Plot
```{r}
#plot
p2.igc <- ggplot(beta_df_between.igc, aes(comparison, between.median.dist)) + 
  geom_boxplot(width = 0.2, outlier.shape = NA)+ 
  geom_half_violin(
      position = position_nudge(x = 0.15, y = 0),
      alpha = 0.25, side = "r"
    ) +
  geom_jitter(aes(comparison, 
                  between.median.dist), 
              alpha=0.25, width = 0.1) + 
  theme_biome_utils() + 
 # ylim(0.6,0.90) +
  theme(axis.text.x = element_text(angle=90, hjust = 0.9)) + 
  labs(title="Median Bray-Curtis Dissimilarity (Between)") + 
  xlab("") + coord_flip() + ggtitle("IGC")
p2.igc
#p2.igc


#ggsave("composition/fig/beta_comparison.pdf", height = 6, width = 6)
#ggsave("composition/fig/beta_comparison.png", height = 6, width = 6, dpi=300) 
```

```{r}
#plot
p2.mpn <- ggplot(beta_df_between.mpn, aes(comparison, between.median.dist)) + 
  geom_boxplot(width = 0.2, outlier.shape = NA)+ 
  geom_half_violin(
      position = position_nudge(x = 0.15, y = 0),
      alpha = 0.25, side = "r"
    ) +
  geom_jitter(aes(comparison, 
                  between.median.dist), 
              alpha=0.25, width = 0.1) + 
  theme_biome_utils() + 
 # ylim(0.6,0.90) +
  theme(axis.text.x = element_text(angle=90, hjust = 0.9)) + 
  labs(title="Median Bray-Curtis Dissimilarity (Between)") + 
  xlab("") + coord_flip() + ggtitle("Metaphlan")
#p2.igc
p2.mpn

#ggsave("composition/fig/beta_comparison.pdf", height = 6, width = 6)
#ggsave("composition/fig/beta_comparison.png", height = 6, width = 6, dpi=300) 
```

```{r fig.width=9, fig.height=10}
p2.igc / p2.mpn
```

From comparisons looks like a highly individual response. Likely due to different starting composition? But we don't know the composition of input community?  

We visualize the personalized response using ordination plot.  
```{r ord, fig.height=4, fig.width=8}
# Ordinate the data
set.seed(42321)
# proj <- get_ordination(pseq, "MDS", "bray")
ord.igc <- ordinate(ps.igc, "MDS", "bray")
p.ord.igc <- plot_ordination(ps.igc, ord.igc, color = "Perturbation",
                shape = "STUDY") +
                geom_point(size = 5, alpha=0.25) + 
  scale_color_manual(values = mycols) + 
  theme_biome_utils() + ggtitle("IGC")
p.ord.igc
#ggsave("composition/fig/beta_ordination.pdf", height = 4, width = 6)
#ggsave("composition/fig/beta_ordination.png", height = 4, width = 6, dpi=300) 
```
This PCoA shows individual microbiota is more similar despite the treatments. 

## Compare beta-diversity  

```{r}
# Ordinate the data
set.seed(42322351)
# proj <- get_ordination(pseq, "MDS", "bray")
ord.mpn <- ordinate(ps.mpn, "MDS", "bray")
p.ord.mpn <- plot_ordination(ps.mpn, ord.mpn, color = "Perturbation",
                shape = "STUDY") +
                geom_point(size = 5, alpha=0.25) + 
  scale_color_manual(values = mycols) + 
  theme_biome_utils() + ggtitle("Metaphlan")
p.ord.mpn
ggsave("02_composition/figs/Beta_metaphln1.pdf", height =6, width = 9)
ggsave("02_composition/figs/Beta_metaphln1.png", height = 6, width = 9, dpi=300)
```

```{r fig.height=4, fig.width=12}
p.ord.igc + p.ord.mpn + plot_layout(guides = "collect")

```

Individuality is preserved.

```{r fig.width=9, fig.height=4}

Combinebetaplot <- ggarrange(p.ord.igc, p.ord.mpn,
                         labels = c("A", "B"),
                         ncol = 2, nrow = 1,
                         common.legend = TRUE, legend = "right")
Combinebetaplot
ggsave("02_composition/figs/beta_igc_metaphln1.pdf", height =6, width = 9)
ggsave("02_composition/figs/beta_igc_metaphln1.png", height = 6, width = 9, dpi=300)

```

#statistical analysis PERMANOVa significance test for group-level differences
```{r}
 

library(vegan)
#for IGC data
permanova <- adonis(t(otu_tb) ~ STUDY,
               data = meta_tabl, permutations=99, method = "bray")

# P-value
print(as.data.frame(permanova$aov.tab)["STUDY", "Pr(>F)"])

#for Metaphlan data
permanova.1 <- adonis(t(otutab) ~ STUDY,
               data = meta_tabl, permutations=99, method = "bray")

# P-value
print(as.data.frame(permanova.1$aov.tab)["STUDY", "Pr(>F)"])


```
